ni/panels/v1/panel_service.proto

```diff
@@ -0,0 +1,146 @@
+  //   - The panel configuration includes a file that was not found.
+  rpc StartPanel(StartPanelRequest) returns (StartPanelResponse);
+
+  // Stop a panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT:
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  rpc StopPanel(StopPanelRequest) returns (StopPanelResponse);
+
+  // Enumerate the panels available in the system, including information about the state of the panels and what values they have.
+  // Status Codes for errors:
+  rpc EnumeratePanels(EnumeratePanelsRequest) returns (EnumeratePanelsResponse);
+
+  // Get a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  // - NOT_FOUND: 
+  //   - The value with the specified identifier was not found.
+  rpc GetValue(GetValueRequest) returns (GetValueResponse);
+
+  // Try to get a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  rpc TryGetValue(TryGetValueRequest) returns (TryGetValueResponse);
+
+  // Set a value for a control on the panel
+  // Status Codes for errors:
+  // - INVALID_ARGUMENT: 
+  //   - The panel identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  //   - The value identifier contains invalid characters. Only alphanumeric characters and underscores are allowed.
+  rpc SetValue(SetValueRequest) returns (SetValueResponse);
+}
+
+message StartPanelRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Configuration for the panel, packed as Any
+  google.protobuf.Any panel_configuration = 2;
+}
+
+message StartPanelResponse {
+  // Location of the panel 
+  string panel_uri = 1; 
+}
+
+message StopPanelRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Reset all storage associated with panel
+  bool reset = 2;
+}
+
+message StopPanelResponse {
+}
+
+message EnumeratePanelsRequest {
+}
+
+message PanelInformation {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Location of the panel
+  string panel_uri = 2;
+
+  // IDs of all of the values associated with the panel
+  repeated string value_ids = 3;
+}
+
+message EnumeratePanelsResponse {
+  // The list of panels available in the system
+  repeated PanelInformation panels = 1;
+}
+
+message GetValueRequest {
+  // Unique ID of the panel
+  string panel_id = 1;
+
+  // Unique ID of value
+  string value_id = 2;
+}
+
+message GetValueResponse {
+  // The value
```

Comment:
Does it make sense to add some documentation about which data types are supported? Or is the list too long to enumerate? Since this is an `Any`, I'm guessing we wrap built in types like `double` into some other message type?

Reply:
Added documentation about supported types.